client/js/src/helpers/api_info.ts

```diff
@@ -60,6 +62,12 @@ export async function process_endpoint(
 	};
 }
 
+export const join_urls = (...urls: string[]): string => {
+	return urls.reduce((base_url, part) => {
+		return `${base_url.replace(/\/+$/, "")}/${part.replace(/^\/+/, "")}`;
+	});
+};
+
```

Comment:
Does `new URL(relative, base)` not handle this?

Reply:
Shouldn't we just throw an error in that case? In node we need that base url so I feel like throwing early would be best. Unless I'm missing something.

Reply:
> we don't know what the base path would be at that point

Just to clarify, we will never know the base path in node right? If it is possible to use the URL class and just propagate that error, then I think that would simplest, if that isn't possible then this fn is fine.